File: backend/src/main/kotlin/ar/edu/unsam/phm/domain/USER_TYPE.kt
package ar.edu.unsam.phm.domain

import ar.edu.unsam.phm.exceptions.BadRequestException

enum class USER_TYPE(val frontName: String) {
    READER("Lector"),
    PUBLISHER("Publicador"),
    ADMIN("Admin");

    companion object {
        fun fromFrontName(value: String): USER_TYPE =
            entries.find { it.frontName.equals(value.trim(), ignoreCase = true) }
                ?: throw BadRequestException("Rol inválido: $value")
    }
}

File: frontend/src/api/domain/USER_TYPE.js
export const USER_TYPE = {
    READER:    { name: "Lector" },
    PUBLISHER: { name: "Publicador" },
    ADMIN:     { name: "Admin" },
}

export const getTypeFromName = (name) => {
    return Object.values(USER_TYPE).find(t => t.name === name)?.id || null
}